File: libs/power-log-parser/src/lib/handlers/shuffle-deck-handler.ts
import { Action, Game, Node, NodeType, ShuffleDeck } from '../models';
import { Regexes } from '../regexes';
import type { ParserState } from '../state/parser-state';

export class ShuffleDeckHandler {
	static Handle(timestamp: string, data: string, state: ParserState, indentLevel: number): boolean {
		const match = Regexes.ActionShuffleDeckRegex.exec(data);
		if (match) {
			const playerId = match[1];

			const shuffleNode = new ShuffleDeck();
			shuffleNode.PlayerId = parseInt(playerId, 10);
			shuffleNode.TimeStamp = timestamp;

			state.UpdateCurrentNode(NodeType.Game, NodeType.Action);
			state.CreateNewNode(new Node(NodeType.ShuffleDeck, shuffleNode, indentLevel, state.Node, data));

			if (state.Node!.Type === NodeType.Game) {
				(state.Node!.Object as Game).AddData(shuffleNode);
			} else if (state.Node!.Type === NodeType.Action) {
				(state.Node!.Object as Action).Data.push(shuffleNode);
			} else {
				throw new Error('Invalid node ' + state.Node!.Type);
			}

			return true;
		}
		return false;
	}
}

File: tools/linux-helpers/hs-converter/HearthstoneReplays/Parser/Handlers/DataHandler/ShuffleDeckHandler.cs
using HearthstoneReplays.Enums;
using HearthstoneReplays.Parser.ReplayData.Entities;
using HearthstoneReplays.Parser.ReplayData.GameActions;
using HearthstoneReplays.Parser.ReplayData;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Action = HearthstoneReplays.Parser.ReplayData.GameActions.Action;

namespace HearthstoneReplays.Parser.Handlers
{
    internal class ShuffleDeckHandler
    {
        public static bool HandleShuffleDeck(DateTime timestamp, string data, ParserState state, int indentLevel)
        {
            var match = Regexes.ActionShuffleDeckRegex.Match(data);
            if (match.Success)
            {
                var playerId = match.Groups[1].Value;

                var shuffleNode = new ShuffleDeck
                {
                    PlayerId = int.Parse(playerId),
                    TimeStamp = timestamp,
                };
                state.UpdateCurrentNode(typeof(Game), typeof(Action));
                state.CreateNewNode(new Node(typeof(ShuffleDeck), shuffleNode, indentLevel, state.Node, data));

                if (state.Node.Type == typeof(Game))
                    ((Game)state.Node.Object).AddData(shuffleNode);
                else if (state.Node.Type == typeof(Action))
                    ((Action)state.Node.Object).Data.Add(shuffleNode);
                else
                    throw new Exception("Invalid node " + state.Node.Type);

                return true;
            }
            return false;
        }
    }
}
